src/qibolab/platforms/abstract.py

```diff
@@ -135,29 +159,53 @@ def reload_settings(self):
         else:
             self.resonator_type = "3D" if self.nqubits == 1 else "2D"
 
-        self.topology = settings["topology"]
-
         self.relaxation_time = settings["settings"]["relaxation_time"]
         self.nshots = settings["settings"]["nshots"]
         self.sampling_rate = settings["settings"]["sampling_rate"]
-
-        # Load native gates
         self.native_gates = settings["native_gates"]
-        self.single_qubit_natives = self.native_gates["single_qubit"]
-        if "two_qubit" in self.native_gates:
-            for gate in self.native_gates["two_qubit"].values():
-                self.two_qubit_natives |= TwoQubitNatives[list(gate)[0]]
-        else:
-            # dummy value to avoid transpiler failure for single qubit devices
-            self.two_qubit_natives = TwoQubitNatives.CZ
 
         # Load characterization settings and create ``Qubit`` and ``Channel`` objects
         for q in settings["qubits"]:
             if q in self.qubits:
                 for name, value in settings["characterization"]["single_qubit"][q].items():
                     setattr(self.qubits[q], name, value)
             else:
-                self.qubits[q] = Qubit(q, **settings["characterization"]["single_qubit"][q])
+                self.qubits[q] = qubit = Qubit(q, **settings["characterization"]["single_qubit"][q])
+                # register channels to qubits when we are using the old format
+                # needed for ``NativeGates`` to work
+                if "qubit_channel_map" in self.settings:
+                    ro, qd, qf, _ = self.settings["qubit_channel_map"][q]
+                    if ro is not None:
+                        qubit.readout = Channel(ro)
+                    if qd is not None:
+                        qubit.drive = Channel(qd)
+                    if qf is not None:
+                        qubit.flux = Channel(qf)
+                # register single qubit native gates to Qubit objects
+                qubit.native_gates = SingleQubitNatives.from_dict(qubit, self.native_gates["single_qubit"][q])
+
+        if len(self.qubits) > 1:
+            for pair in settings["topology"]:
+                pair = tuple(sorted(pair))
+                if pair not in self.pairs:
+                    self.pairs[pair] = QubitPair(self.qubits[pair[0]], self.qubits[pair[1]])
+
+        # Load native two-qubit gates
+        if "two_qubit" in self.native_gates:
```

Comment:
I would leave the check for now because the single qubit runcards may not contain the `two_qubit` field at all. However we can review this when we cleanup platforms, when the platform will be disentangled from the runcard anyway.

Reply:
Ok: I would simplify the format and always keep the key, even for the single qubit (just leaving it empty).

But, at some point, we will have a `dataclass` for the runcard. So, maybe we can review at that point :)